### ISSUE TITLE
BatchNormalization gives Warning

### ISSUE THREAD

**Issue by skjerns** [kuza55/keras-extras#17]
When I use a model that uses BatchNorm, I get a lot (~100) of warnings in the following form:

```
WARNING:tensorflow:Tried to colocate gradients/tower_0/sequential_1/batch_normalization_5/moments/sufficient_statistics/count_grad/Rank with an op tower_0/sequential_1/batch_normalization_5/moments/sufficient_statistics/count that had a different device: /device:CPU:0 vs /device:GPU:0. Ignoring colocation property.
```

After googleing I found out it maybe has to do with some functions not being implemented on the GPU (yet).

How would it be possible to avoid those warnings? Or to fix the bug?
Can I just ignore the warning? After all it seems to train okayish.

**Comment by kuza55** [comment]
This is safe to ignore, but does sound like it may impact speed.

The error message seems to imply that the gradients are being allocated on the CPU, whereas the operation is on the GPU, which would certainly slow down training if gradients are needlessly being written to main memory.
